src/pubsub.c

```diff
@@ -578,17 +578,30 @@ void punsubscribeCommand(client *c) {
     if (clientTotalPubSubSubscriptionCount(c) == 0) c->flags &= ~CLIENT_PUBSUB;
 }
 
+int pubsubPublishMessageByType(robj *channel, robj *message, int sharded) {
```

Comment:
Because the name contains "ByType", it's intuitive to think that the 3rd argument is a type rather than a boolean. We can solve this by defining and using macros instead of 0 and 1:

```C
#define PUBSUB_TYPE_CLASSIC 0
#define PUBSUB_TYPE_SHARDED 1
```

... and rename `int sharded` to `int pubsub_type`

Reply:
Btw, I can't see that this refactoring is necessary. We could just call the existing functions from module.c, can't we?

Reply:
since this argument is mandatory, i don't think we need to mention that in the function name.
i.e. it could just be `pubsubPublishMessage` that takes a `sharded` flag.
it would be nice to add a top comment above the function describing what it does and it's argument.
maybe it should actually be called `pubsubPublishMessageAndPropagateToCluster` 8-)

Reply:
@zuiderkwast yes, but then there's gonna be more code repetition than I think is acceptable (4 calls to pubsubPublishMessage/pubsubPublishMessageShard and 4 calls to clusterPropagatePublish/clusterPropagatePublishShard)

Reply:
Well, we don't want more code repetition. :-) What about some refactoring to remove code repetition? Let's say we go for the `int sharded` 3rd arg in all functions? That is

* replace `pubsubPublishMessage` and `pubsubPublishMessageShard` with a `pubsubPublishMessage(channel, message, sharded)` and
* add `pubsubPublishMessageAndPropagateToCluster(channel, message, sharded)`

Reply:
i'll try, let's see how it looks like (same for `clusterPropagatePublishShard`)

Reply:
looking good